File: platform/frontend/src/api/deploymenthealth.ts
import i18n from '../i18n'
import { authFetch } from './auth'

const API_BASE_URL = import.meta.env.VITE_API_URL || 'http://localhost:5000'
const t = (key: string) => i18n.t(key)

export interface DeploymentHealthConfig {
  id: string
  projectId: string
  deploymentUrl: string
  status: string
  monitoringEnabled: boolean
  checkIntervalSeconds: number
  errorRateThreshold: number
  latencyThresholdMs: number
  autoRollbackEnabled: boolean
  totalChecks: number
  successfulChecks: number
  failedChecks: number
  currentErrorRate: number
  avgResponseTimeMs: number
  p95ResponseTimeMs: number
  p99ResponseTimeMs: number
  uptimePercentage: number
  rollbackCount: number
  lastCheckAt: string | null
}

export interface HealthStats {
  status: string
  uptimePercentage: number
  totalChecks: number
  successfulChecks: number
  failedChecks: number
  currentErrorRate: number
  avgResponseTimeMs: number
  p95ResponseTimeMs: number
  p99ResponseTimeMs: number
  rollbackCount: number
  lastCheckAt: string | null
}

export interface HealthEvent {
  timestamp: string
  status: string
  responseTimeMs: number
  error: string | null
}

export interface IncidentRecord {
  startedAt: string
  resolvedAt: string | null
  type: string
  description: string
}

export async function getHealthConfig(projectId: string): Promise<DeploymentHealthConfig> {
  const response = await authFetch(`${API_BASE_URL}/api/deployment-health/config/${projectId}`)
  if (!response.ok) {
    const error = await response.json().catch(() => ({}))
    throw new Error(error.error || t('api.error.healthConfigLoad'))
  }
  return response.json()
}

export async function updateHealthConfig(projectId: string, config: {
  deploymentUrl?: string
  monitoringEnabled: boolean
  checkIntervalSeconds: number
  errorRateThreshold: number
  latencyThresholdMs: number
  autoRollbackEnabled: boolean
}): Promise<DeploymentHealthConfig> {
  const response = await authFetch(`${API_BASE_URL}/api/deployment-health/config/${projectId}`, {
    method: 'PUT',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify(config),
  })
  if (!response.ok) {
    const error = await response.json().catch(() => ({}))
    throw new Error(error.error || t('api.error.healthConfigUpdate'))
  }
  return response.json()
}

export async function getHealthStats(projectId: string): Promise<HealthStats> {
  const response = await authFetch(`${API_BASE_URL}/api/deployment-health/stats/${projectId}`)
  if (!response.ok) {
    const error = await response.json().catch(() => ({}))
    throw new Error(error.error || t('api.error.healthStatsLoad'))
  }
  return response.json()
}

export async function getHealthEvents(projectId: string): Promise<HealthEvent[]> {
  const response = await authFetch(`${API_BASE_URL}/api/deployment-health/events/${projectId}`)
  if (!response.ok) {
    const error = await response.json().catch(() => ({}))
    throw new Error(error.error || t('api.error.healthEventsLoad'))
  }
  return response.json()
}

export async function getIncidents(projectId: string): Promise<IncidentRecord[]> {
  const response = await authFetch(`${API_BASE_URL}/api/deployment-health/incidents/${projectId}`)
  if (!response.ok) {
    const error = await response.json().catch(() => ({}))
    throw new Error(error.error || t('api.error.healthIncidentsLoad'))
  }
  return response.json()
}

File: platform/backend/AiDevRequest.API/Entities/DeploymentHealth.cs
using System.ComponentModel.DataAnnotations;

namespace AiDevRequest.API.Entities;

public class DeploymentHealth
{
    public Guid Id { get; set; } = Guid.NewGuid();

    [Required]
    [MaxLength(100)]
    public required string UserId { get; set; }

    /// <summary>
    /// Dev request / project this health check belongs to
    /// </summary>
    public Guid DevRequestId { get; set; }

    /// <summary>
    /// Deployment URL being monitored
    /// </summary>
    [Required]
    [MaxLength(500)]
    public string DeploymentUrl { get; set; } = "";

    /// <summary>
    /// Current status: up, degraded, down
    /// </summary>
    [MaxLength(20)]
    public string Status { get; set; } = "unknown";

    /// <summary>
    /// Whether monitoring is enabled for this deployment
    /// </summary>
    public bool MonitoringEnabled { get; set; } = true;

    /// <summary>
    /// Health check interval in seconds
    /// </summary>
    public int CheckIntervalSeconds { get; set; } = 60;

    /// <summary>
    /// Error rate threshold (0.0 - 1.0) to trigger rollback
    /// </summary>
    public double ErrorRateThreshold { get; set; } = 0.1;

    /// <summary>
    /// Response time P95 threshold in ms to mark as degraded
    /// </summary>
    public int LatencyThresholdMs { get; set; } = 5000;

    /// <summary>
    /// Total number of health checks performed
    /// </summary>
    public int TotalChecks { get; set; } = 0;

    /// <summary>
    /// Number of successful checks
    /// </summary>
    public int SuccessfulChecks { get; set; } = 0;

    /// <summary>
    /// Number of failed checks
    /// </summary>
    public int FailedChecks { get; set; } = 0;

    /// <summary>
    /// Current error rate (0.0 - 1.0)
    /// </summary>
    public double CurrentErrorRate { get; set; } = 0;

    /// <summary>
    /// Average response time in ms (P50)
    /// </summary>
    public double AvgResponseTimeMs { get; set; } = 0;

    /// <summary>
    /// P95 response time in ms
    /// </summary>
    public double P95ResponseTimeMs { get; set; } = 0;

    /// <summary>
    /// P99 response time in ms
    /// </summary>
    public double P99ResponseTimeMs { get; set; } = 0;

    /// <summary>
    /// Uptime percentage (0.0 - 100.0)
    /// </summary>
    public double UptimePercentage { get; set; } = 100;

    /// <summary>
    /// Number of auto-rollbacks triggered
    /// </summary>
    public int RollbackCount { get; set; } = 0;

    /// <summary>
    /// Whether auto-rollback is enabled
    /// </summary>
    public bool AutoRollbackEnabled { get; set; } = false;

    /// <summary>
    /// Version ID of last known good deployment
    /// </summary>
    public Guid? LastGoodVersionId { get; set; }

    /// <summary>
    /// JSON array of recent health check events [{timestamp, status, responseTimeMs, error}]
    /// </summary>
    public string? HealthEventsJson { get; set; }

    /// <summary>
    /// JSON array of incident records [{startedAt, resolvedAt, type, description}]
    /// </summary>
    public string? IncidentsJson { get; set; }

    public DateTime? LastCheckAt { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
}
